app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java

```diff
@@ -176,124 +157,106 @@
   private View actionModeView;
   private FastScroller fastScroller;
   private CustomFileObserver customFileObserver;
-  private DataUtils dataUtils;
-  private boolean isEncryptOpen =
-      false; // do we have to open a file when service is begin destroyed
-  private HybridFileParcelable
-      encryptBaseFile; // the cached base file which we're to open, delete it later
-  private int ordinalValue;
-
-  /** a list of encrypted base files which are supposed to be deleted */
-  private ArrayList<HybridFileParcelable> encryptBaseFiles = new ArrayList<>();
 
   // defines the current visible tab, default either 0 or 1
   // private int mCurrentTab;
 
-  /*
-   * boolean identifying if the search task should be re-run on back press after pressing on
-   * any of the search result
-   */
-  private boolean mRetainSearchTask = false;
-
-  /** For caching the back button */
-  private LayoutElementParcelable back = null;
-
-  private int dragAndDropPreference;
+  private MainFragmentViewModel mainFragmentViewModel;
```

Comment:
Why do you keep this? In some cases `mainFragmentViewModel` will be null when `getMainFragmentViewModel()` wouldn't be.

Reply:
A null check is there for all public methods or `getMainFragmentViewModel()` This class level variable is defined for private methods so that I don't need to create it again and again.

Reply:
> I don't need to create it again and again.

Create what? The `ViewModelProvider`?

Reply:
Why don't you want to create it again and again?

Reply:
Yes, I don't need to declare in each method that's why it's at class level

Reply:
> Yes, I don't need to declare in each method that's why it's at class level

But you lose the guarantee that it won't be null. You'd have to check that for each usage.

Reply:
For private members in MainFragment - it's initialized in oncreateview so it'll not be null. In public methods exposed by MainFragment I'm making that check everywhere until we have a better approach. We shouldn't be exposing fragment methods like this. These are last minute changes so I didn't want to make much changes than required.
It's either this or I put try catch in each method, only to replace them later by a better approach.

Reply:
> In public methods exposed by MainFragment I'm making that check everywhere until we have a better approach. 

Fixing [this](https://github.com/TeamAmaze/AmazeFileManager/pull/2551#pullrequestreview-676876595) should fix this as no more null check would be required. Especially inside the Fragment.